ext/vscode/src/commands/down.ts

```diff
@@ -2,52 +2,51 @@
 // Licensed under the MIT License.
 
 import * as vscode from 'vscode';
-import * as path from 'path';
 import { IActionContext } from "@microsoft/vscode-azext-utils";
-import { localize } from '../localize';
-import { quickPickWorkspaceFolder } from '../utils/quickPickWorkspaceFolder';
-import { getAzDevTerminalTitle, pickAzureYamlFile, selectApplicationTemplate, showReadmeFile } from './cmdUtil';
+import { getAzDevTerminalTitle, getWorkingFolder, } from './cmdUtil';
 import { createAzureDevCli } from '../utils/azureDevCli';
 import { executeAsTask } from '../utils/executeAsTask';
 import { TelemetryId } from '../telemetry/telemetryId';
 import { AzureDevCliApplication } from '../views/workspace/AzureDevCliApplication';
 import { isTreeViewModel, TreeViewModel } from '../utils/isTreeViewModel';
+import { localize } from '../localize';
 
 export async function down(context: IActionContext, selectedItem?: vscode.Uri | TreeViewModel): Promise<void> {
     const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliApplication>().context.configurationFile : selectedItem;
+    const workingFolder = await getWorkingFolder(context, selectedFile);
 
-    let folder: vscode.WorkspaceFolder | undefined = (selectedFile ? vscode.workspace.getWorkspaceFolder(selectedFile) : undefined);
-    if (!folder) {
-        folder = await quickPickWorkspaceFolder(context, localize('azure-dev.commands.cli.init.needWorkspaceFolder', "To run '{0}' command you must first open a folder or workspace in VS Code", 'down'));
-    }
+    const confirmPrompt = localize('azure-dev.commands.cli.down.confirm-prompt', "Are you sure you want to delete all application's Azure resources?");
+    const confirmAck = localize('azure-dev.commands.cli.down.confirm-ack', "Delete resources");
 
-    const azureCli = await createAzureDevCli(context);
-    let command = azureCli.commandBuilder
-        .withArg('down');
-    let workingDir = folder.uri;
+    await context.ui.showWarningMessage(confirmPrompt, { modal: true }, { title: confirmAck });
 
-    const azureYamlFile = selectedFile ?? await pickAzureYamlFile(context);
-    if (azureYamlFile) {
-        // Workspace has already been initialized, no need to specify a template
-        workingDir = vscode.Uri.file(path.dirname(azureYamlFile.fsPath));
-    } else {
-        const templateUrl = await selectApplicationTemplate(context);
-        command = command.withNamedArg('-t', {value: templateUrl, quoting: vscode.ShellQuoting.Strong});
-    }
+    const choices = [
+        {
+            label: localize("azure-dev.commands.cli.down.no-purge", "Do not purge"),
+            data: false
+        },
+        {
+            label: localize("azure-dev.commands.cli.down.purge", "Permanently delete (purge)"),
+            data: true
+        }
+    ];
+    const purgeChoice = await context.ui.showQuickPick(choices, {
```

Comment:
The UX here of going from a modal to a quick pick is really awkward. Should just make this a modal too.

Reply:
I feel like multiple modals back to back feels sort of yucky? I'd probably opt for approach 2 unless you have reservations.

Reply:
I'm fine with that. Can you provide text for the prompt and buttons?

Reply:
Wait, I'm now confused after looking at this in VS Code again. When do these modals show up? If I run `down` from azure.yaml or command palette, we currently just shell out to the embedded terminal.

Reply:
A general goal I have is to do more prompting in VSCode and less in the terminal. That's the experience users are accustomed to. For now it's really tough to chase the CLI, but as things stabilize, I'd like to shift more and more prompting inward, with arguments to pass to the CLI to suppress the prompt in the CLI.

In this particular case, when you do `azd down`, we currently (in the public builds) ask "Are you sure?", and then the CLI asks if you also want to purge. With the changes as-currently-written, it's a modal "Are you sure?" and then a quick pick purge or not.

For approach 2 we basically just fold those into one modal question. "Are you sure? Do you wanna purge?" [Yes and purge] [Yes and don't purge] [Cancel]

Reply:
Okay, phew. I thought I was missing something where we were already using the command palette for this. All good.

Yes -  we definitely want to use the UI over the terminal where possible.

A couple of questions:
1. Are we able to show the resource count in the modal? Similar to how it shows in the terminal.
![image](https://user-images.githubusercontent.com/8949415/229230289-220363aa-d02c-4759-a8dc-81ca63520e41.png)
1. Are key vaults the only thing that would need to be purged?

 I'm a little worried that people won't know the difference between purge and soft delete but it could be something like this?

"`azd down` will delete xx resources, are you sure you want to continue" 
1. "Yes, I would like to continue and purge all resources"
1. "Yes, I would like to continue and soft delete resources"
1.  "Cancel"

Reply:
We could show a resource count but it would require some additional work, which I'm not sure I can get done by the 5th.

Reply:
No worries - Are key vaults the only thing that would need to be purged?


How do you feel about

"azd down will delete this application's Azure resources and deployments, are you sure you want to continue"

"Yes, I would like to continue and purge all resources"
"Yes, I would like to continue and soft delete resources"
"Cancel"

Reply:
That seems great!

Reply:
Ok, here's the final version:
![image](https://user-images.githubusercontent.com/36966225/229573319-7a72707c-bf04-4d76-923e-ce2fb3d80177.png)